econml/automated_ml.py

```diff
@@ -0,0 +1,360 @@
+    ``create_workspace`` is set true, a new workspace is created
+    for the user. If ``create_workspace`` is set true, a new workspace is
+    created for the user.
+
+    Parameters
+    ----------
+
+    create_workspace: Boolean, optional, default False
+       If set to true, a new workspace will be created if the specified
+       workspace does not exist.
+
+    create_resource_group: Boolean, optional, default False
+       If set to true, a new resource_group will be created if the specified
+       resource_group does not exist.
+
+    workspace_region: String, optional
+       Region of workspace, only necessary if create_new is set to true and a
+       new workspace is being created.
+
+    subscription_id: String, required
+       Definition of a class that will serve as the parent class of the
+       AutomatedMLMixin. This class must inherit from _BaseDMLCateEstimator.
+
+    resource_group: String, required
+       Name of resource group of workspace to be created or set.
+
+    workspace_name: String, required
+       Name of workspace of workspace to be created or set.
+    """
+    try:
+        ws = Workspace(subscription_id=subscription_id, resource_group=resource_group,
+                       workspace_name=workspace_name, auth=auth)
+        # write the details of the workspace to a configuration file to the notebook library
+        ws.write_config()
+        print("Workspace configuration has succeeded.")
+    except ProjectSystemException:
+        if(create_workspace):
+            if(create_resource_group):
+                print("Workspace not accessible. Creating a new workspace and \
+                resource group.")
+                ws = Workspace.create(name=workspace_name,
+                                      subscription_id=subscription_id,
+                                      resource_group=resource_group,
+                                      location=workspace_region,
+                                      create_resource_group=create_resource_group,
+                                      sku='basic',
+                                      auth=auth,
+                                      exist_ok=True)
+                ws.get_details()
+            else:
+                print("Workspace not accessible. Set \
+                create_resource_group = True and run again to create a new \
+                workspace and resource group.")
+        else:
+            print("Workspace not accessible. Set create_workspace = True \
+            to create a new workspace.")
+
+
+def addAutomatedML(baseClass):
+    """
+    Add AutomatedMLMixin to specified base class.
+
+    In addition to a mixin that creates automatedML models from AutoMLConfig
+    objects, another mixin is added specifying modifications to the
+    AutoMLConfig. The business logic that this mixin runs is detailed below:
+
+    If the baseClass is of type NonParamDMLCateEstimator
```

Comment:
1. I don't see any logic that actually does this - am I missing something?
2. Both here and for what you say about linearity below, only the final stage model has this requirement; the first stage models do not need to support weights (or be linear, in the other case)

Reply:
Yes I apologize for this, this was an old implementation, now, the way that users can specify whether they want to constrain their model space happens in the initialization of an EconAutoMLConfig object.